pkg/tests/ossm/prometheus_scoped_scraping_test.go

```diff
@@ -0,0 +1,260 @@
+		})
+
+		t.NewSubTest("when removing a namespace from existing SMMR").Run(func(t test.TestHelper) {
+			ns := generateNamespace()
+			anotherNs := generateNamespace()
+
+			t.Cleanup(func() {
+				oc.DeleteNamespace(t, ns, anotherNs)
+				restoreDefaultSMMR(t)
+			})
+
+			t.LogStepf("Create namespace (%s,%s) and add it into SMMR", ns, anotherNs)
+			oc.CreateNamespace(t, ns, anotherNs)
+			updateDefaultSMMRWithNamespace(t, ns, anotherNs)
+
+			t.LogStepf("Update SMMR with only %s", ns)
+			updateDefaultSMMRWithNamespace(t, ns)
+
+			t.LogStepf("Look for %s in prometheus ConfigMap", ns)
+			testPrometheusConfigWithAsserts(t, assertConfigMapContainsNamespace(ns), assertConfigMapDoesNotContainNamespace(anotherNs))
+		})
+
+		t.NewSubTest("when the default SMMR with no member").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				restoreDefaultSMMR(t)
+			})
+
+			start := time.Now()
+			t.LogStepf("Update default SMMR with no member")
+			updateDefaultSMMRWithNamespace(t)
+
+			checkConfigurationReloadingTriggered(t, start)
+			checkPermissionError(t)
+		})
+
+		t.NewSubTest("when the default SMMR with nonexistent namespace").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				restoreDefaultSMMR(t)
+			})
+
+			t.LogStepf("Update default SMMR with nonexistent member")
+
+			ns := generateNamespace()
+
+			s := buildSMMR(ns)
+
+			t.LogStepf("Update SMMR %s", s)
+			oc.ApplyString(t, meshNamespace, s)
+
+			testPrometheusConfigWithAsserts(t, assertConfigMapDoesNotContainNamespace(ns))
+			checkPermissionError(t)
+		})
+
+		t.NewSubTest("query istio_request_total").Run(func(t test.TestHelper) {
+			ns := "bookinfo"
+			t.Cleanup(func() {
+				oc.RecreateNamespace(t, ns)
+			})
+
+			t.LogStep("Install bookinfo")
+			app.InstallAndWaitReady(t, app.Bookinfo(ns))
+
+			count := 10
+			t.LogStepf("Generate %d requests to product page", count)
+			productPageURL := app.BookinfoProductPageURL(t, meshNamespace)
+			for i := 0; i < count; i++ {
+				curl.Request(t, productPageURL, nil)
+			}
+
+			t.LogStep(`Check if the "istio_request_total metric is in Prometheus"`)
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.Exec(t,
+					prometheusPodSelector,
+					"prometheus-proxy",
+					"curl localhost:9090/api/v1/query --data-urlencode 'query=istio_requests_total'",
+					assert.OutputContains(
+						`"__name__":"istio_requests_total"`,
+						`Found the "istio_request_total" metric`,
+						`Expected to find the "istio_request_total" metric, but found none`,
+					),
+				)
+			})
+		})
+
+		t.NewSubTest("when removing SMMR").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				restoreDefaultSMMR(t)
+			})
+
+			checks := []common.CheckFunc{assertConfigMapContainsNamespace(meshNamespace)}
+			for _, ns := range []string{namespaces.Bar, namespaces.Bookinfo, namespaces.Foo, namespaces.Legacy} {
+				checks = append(checks, assertConfigMapDoesNotContainNamespace(ns))
+			}
```

Comment:
Does it make sense to create this array? Why not simply call these assert functions in line 207? It would make the test a bit more readable and simpler.

Reply:
.. and if you really don't like to see `assertConfigMapDoesNotContainNamespace` repeated four times, you could change it to accept `namespaces ...string` instead of just one namespace.